docs/site/migration/models/mixins.md

```diff
@@ -6,8 +6,805 @@ sidebar: lb4_sidebar
+**mixins** section in **common/models/note.json** to indicate which mixins
+should be applied to it.
+
+{% include code-caption.html content="common/models/note.json" %}
+
+```json
+{
+  "name": "Note",
+  "properties": {
+    "title": {
+      "type": "string",
+      "required": true
+    },
+    "content": {
+      "type": "string"
+    }
+  },
+
+  "mixins": {
+    "FindByTitle": {
+      "returnArgumentName": "notes"
+    },
+
+    "Category": true
+  }
+}
+```
+
+Specifying an options object for `FindByTitle` is the same as specifying a value
+of **true** as it will apply the **findByTitle.js** custom model method/remote
+method mixin to the `Note` model. A value of **false** will not apply the mixin.
+
+## LoopBack 4 Approach to Creating A Custom Model Method Mixin And Remote Model Method Mixin
+
+As mentioned in the previous section, the
+[Add a New Model Method And a New Endpoint](./methods.md#add-a-new-model-method-and-a-new-endpoint)
+section of the [Migrating custom model methods](./methods.md) document explains
+how a LoopBack 3 developer can define a custom model method named `findByTitle`
+on the `Note` model, and define a remote method to make it available as a new
+endpoint. It then shows how a LoopBack 4 developer can implement a `findByTitle`
+method on the `NoteRepository` and on the `NoteController` to accomplish the
+same thing.
+
+In this section, we will show how a LoopBack 4 developer can define two mixins (
+a repository mixin and a controller mixin) to add a `findByTitle` method to
+`NoteRepository` and `NoteController` respectively.
+
+In LoopBack 4, a developer is able to create a repository and controller method
+mixin by:
+
+- defining a common interface for both mixin class factory functions
+- placing the mixin class factory functions in separate files
+- generating a repository and controller using the CLI as usual
+- adjusting the repository and controller files to make use of its respective
+  mixin class factory function
+
+### Defining A Common Interface For The findByTitle Method
+
+Let's define a common interface `FindByTitleInterface` in
+**src/mixins/findByTitleInterface.ts**.
+
+{% include code-caption.html content="src/mixins/findByTitleInterface.ts" %}
+
+```ts
+import {Model} from '@loopback/repository';
+
+export interface FindByTitleInterface<M extends Model> {
+  findByTitle(title: string): Promise<M[]>;
+}
+```
+
+### Defining A Repository Mixin Class Factory Function
+
+In **src/mixins/findByTitleRepositoryMixin.ts**, let's define the mixin class
+factory function `FindByTitleRepositoryMixin` which adds the `findByTitle`
+method to any repository.
+
+{% include code-caption.html content="src/mixins/findByTitleRepositoryMixin.ts" %}
+
+```ts
+import {Constructor} from '@loopback/context';
+
+import {Model} from '@loopback/repository';
+import {FindByTitleInterface} from './findByTitleInterface';
+
+export function FindByTitleRepositoryMixin<
+  M extends Model,
+  T extends Constructor<any>
```

Comment:
Is it possible to clarify these generic arguments to make it more clear that T is expected to be a class implementing `CrudRepository<M>`? 

Ideally, this should be expressed in a way that allows the TypeScript compiler to verify the contract at compile time. If that's not possible or too difficult, then at least add a short comment to explain what is `T` for. 

I also think that `R` would be a better name (`R` as a shortcut for `Repository`).